src/state/cached_state.rs

```diff
@@ -126,6 +128,23 @@ impl<T: StateReader, C: ContractClassCache> CachedState<T, C> {
     }
 }
 
+impl<T: StateReader, C: ContractClassCache> Clone for CachedState<T, C> {
```

Comment:
We should look for a way of avoiding this clone implementation.

Reply:
Replaced it with `clone_for_testing`